### PR TITLE
BPR-3467: add bankAccount JSON field to handle bankAccount structure

### PR DESCRIPTION
**Pull request by dawidsappich**
- handle insert and loading of new column which is defined in database with datatype JSON
- refactor string representation of bankaccount
- add abaRoutingNumber to string representation of bankAccount
- handle bankAccount and abaRoutingNumber in state object

### CODE REVIEW

File: src/client/api/BankAccounts.js [hunk 1]
```diff
@@ -11,21 +9,15 @@ class BankAccounts
...
-    static getStringRepresentation(i18n, iban, bankgiro, plusgiro, isrNumber)
+    static getStringRepresentation(i18n, bankAccount)
     {
         // console.log("BankAccounts.getPosition: ", iban, bankgiro, plusgiro, isrNumber);
         let values = []
-        if (iban) values.push(`${i18n.getMessage('Labels.iban')}: ${iban}`);
-        if (bankgiro) values.push(`${i18n.getMessage('Labels.bankgiro')}: ${bankgiro}`);
-        if (plusgiro) values.push(`${i18n.getMessage('Labels.plusgiro')}: ${plusgiro}`);
-        if (isrNumber) values.push(`${i18n.getMessage('Labels.isr')}: ${isrNumber}`);
+        if (bankAccount.iban) values.push(`${i18n.getMessage('Labels.iban')}: ${bankAccount.iban}`);
```
[the remark this answers is not in the document]
Comment (nczeranka) [reply src/client/api/BankAccounts.js:16]: See https://opuscapita.atlassian.net/browse/BPR-3785

File: src/client/api/BankAccounts.js [hunk 2]
```diff
@@ -11,21 +9,15 @@ class BankAccounts
         return bankAccount.accountNumber + bankAccount.bankgiro + bankAccount.plusgiro + bankAccount.isrNumber;
```
[the remark this answers is not in the document]
Comment (nczeranka) [reply src/client/api/BankAccounts.js:9]: Will be worked on via https://opuscapita.atlassian.net/browse/BPR-3785

**Comment by dawidsappich** [comment]
> Possible problems that should be checked:
> 
>     * Generation of the PDF-Representation
> 
>     * Enpoints provide correct JSON - including BankAccount as JSON
> 
>     * Event emitting with correct JSON - including the BankAccount as JSON
> 
> 
> Besides of it: the general BankAccount handling will be revised via https://opuscapita.atlassian.net/browse/BPR-3785

- Generation of the PDF-Representation
  - `invoiceDocDefinition.js` uses iban, bankgiro etc. Definition is used with `pdfmake` to generate a pdf. These infos are rendered as supplier info in the PDF. This has to be adressed in [Sales-Invoice: Revise BankAccount handling](https://opuscapita.atlassian.net/browse/BPR-3785) Example of an PDF-Preview. 
pdf-invoice
- Enpoints provide correct JSON - including BankAccount as JSON
  - all relevant endpoints are handled with `salesinvoice#dataCleansing`. bankAccount as JSON object is ensured
  - **saving** an invoice generates a PDF in Blob service. PDF generation is not handling with bankAcocunt right now
  - **updating** an invoice generates a PDF in Blob service. PDF generation is not handling with bankAcocunt right now

- Event emitting with correct JSON - including the BankAccount as JSON
  - saving, updating and deleting an invoice emits the corresponding events with the invoice. Loading the corresponding invoice is processed in conjunction with `salesinvoice#dataCleansing`

File: src/client/api/SalesInvoices.js [hunk 3]
```diff
@@ -193,6 +193,8 @@ class SalesInvoices
                 item.netPriceCopyOnLoad = item.netPrice;
             });
         }
+        // map JSON Data as JSON Object
+        if(invoice && invoice.bankAccount) invoice.bankAccount = JSON.parse(invoice.bankAccount);
```
Comment (nczeranka) [inline src/client/api/SalesInvoices.js:197]: Now that the a JSON object is provided, JSON.parse will run into an error.

Comment (dawidsappich) [reply src/client/api/SalesInvoices.js]: i missed that one. i have removed the snippet because it is sufficient that the backend is doing the JSON parsing

**Pull request merged**